File: app/src/main/java/com/ladsoft/bakingapp/mvp/presenter/Presenter.java
package com.ladsoft.bakingapp.mvp.presenter;

import java.lang.ref.WeakReference;

public abstract class Presenter<V> {
    private WeakReference<V> view;

    public void attachView(V view) {
        this.view = new WeakReference<>(view);
    }

    public void detachView() {
        this.view = null;
    }

    protected V getView() {
        return this.view.get();
    }

    protected boolean isViewAttached() {
        return this.view != null && this.view.get() != null;
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/mvp/presenter/Presenter.kt
package com.ladsoft.bakingapp.mvp.presenter

abstract class Presenter<V> {

    open var view: V? = null

    open fun attachView(view: V) {
        this.view = view
    }

    open fun detachView() {
        this.view = null
    }

    fun isViewAttached(): Boolean {
        return this.view != null
    }
}

